### PR TITLE
migrate to hatchling and nox

### PR DESCRIPTION
**Pull request by gotmax23**
TODO:
- [x] Add publish and bump sessions
- [x] Apply changes from https://github.com/ansible-community/antsibull-docs-parser/pull/3
- [x] Update README badges

### CODE REVIEW

**Comment by felixfontein** [comment]
@gotmax23 I think this PR can use an update :)

**Review by felixfontein — APPROVED** [review]

**Comment by gotmax23** [comment]
:tada:

File: pyproject.toml [hunk 1]
```diff
@@ -3,62 +3,109 @@
...
-pydantic = ">= 1.0.0 < 2.0.0"
-PyYAML = "*"
-semantic_version = "*"
-sh = ">= 1.0.0 < 2.0.0"
-# 0.5.0 introduces dict_config
-twiggy = ">= 0.5.0"
+[project.optional-dependencies]
+codeqa = [
+    "flake8 >= 6.0.0",
+    "pylint ~= 2.15.7",
```
Comment (felixfontein) [inline pyproject.toml:69]: Why not
```suggestion
    "pylint >= 2.15.7",
```
? Or if you want to keep the old behavior:
```suggestion
    "pylint ~= 2.15",
```

File: pyproject.toml [hunk 2]
```diff
@@ -3,62 +3,109 @@
...
+    "cryptography",
+    "pytest",
+    "pytest-asyncio >= 0.20",
+    "pytest-cov",
+    "pytest-error-for-skips",
+]
+typing = [
+    "mypy",
+    # https://github.com/facebook/pyre-check/issues/398
+    "pyre-check ~= 0.9.17",
```
Comment (felixfontein) [inline pyproject.toml:86]:
```suggestion
    "pyre-check >= 0.9.17",
```

**Comment by gotmax23** [comment]
I created https://github.com/ansible-community/antsibull-core/pull/46 to fix the version constraints.

**Pull request merged**